File: api/src/main/java/com/wbrawner/budgetserver/user/UserRepository.java
package com.wbrawner.budgetserver.user;

import org.springframework.data.repository.PagingAndSortingRepository;

import java.util.List;
import java.util.Optional;

public interface UserRepository extends PagingAndSortingRepository<User, String> {
    Optional<User> findByUsername(String username);

    Optional<User> findByUsernameAndPassword(String username, String password);

    List<User> findByUsernameContains(String username);

    Optional<User> findByEmail(String email);
}

File: src/main/kotlin/com/wbrawner/budgetserver/user/UserRepository.kt
package com.wbrawner.budgetserver.user

import org.springframework.data.repository.PagingAndSortingRepository
import java.util.*

interface UserRepository: PagingAndSortingRepository<User, Long> {
    fun findByName(username: String): Optional<User>
    fun findByNameContains(username: String): List<User>
    fun findByEmail(email: String): Optional<User>
}